modules/bmc/redfish.rb

```diff
@@ -0,0 +1,235 @@
+require 'redfish_client'
+require 'bmc/base'
+require 'bmc/redfish/dell'
+require 'bmc/redfish/hpe'
+
+module Proxy
+  module BMC
+    class Redfish < Base
+      include Proxy::Log
+      include Proxy::Util
+
+      def initialize(args)
+        @redfish_verify_ssl = Proxy::BMC::Plugin.settings.bmc_redfish_verify_ssl
+        # enforce SSL verification as a default, if the value isn't set in the config
+        # and bail if it's set to something nonsensical
+        @redfish_verify_ssl = true if @redfish_verify_ssl.nil?
+        raise ::Proxy::Error::ConfigurationError.new("bmc_redfish_verify_ssl must be boolean (true/false)") unless [true, false].include?(@redfish_verify_ssl)
```

Comment:
We actually have a mechanism now to provide default values to settings via `default_setting` method, see `modules/dhcp/dhcp_plugin.rb` as an example. But there is no validation API.

Reply:
now with 100% more default_setting